appshell/appshell_extensions.cpp

```diff
@@ -765,8 +783,53 @@ class ProcessMessageDelegate : public ClientHandler::ProcessMessageDelegate {
 			//  0: int32 - callback id
 
 			responseArgs->SetString(2, GetSystemUniqueID());
-		}
-		else {
+		} 
+        else if (message_name == "ReadDirWithStats") {
```

Comment:
So here is the thing. In my testing what I figured out was `readDir` leads to lot of shell calls. Here is what happens for `readDir`. First we get plain contents of a directory and for every file entry, we then make a `stat` call to shell. With our Brackets repository open, this led to close to 33,000 shell calls. So if all we want is to get directory contents and file stats we could very well do that in a single shell call. Using this has reduced the no of shell calls to 5000 for readDir, for Brackets repo. So I guess this should pretty much be made mainstream. For now, I have just kept it added and it another PR on the Brackets side, I would replace "ReadDir" shell call with "ReadDirWithStats". Do you think it makes sense to remove this here and have another PR for this?